drivers/sensor/lis2dw12/lis2dw12.c

```diff
@@ -483,11 +519,57 @@ static int lis2dw12_init(const struct device *dev)
 #warning "LIS2DW12 driver enabled without any devices"
 #endif
 
+#ifdef CONFIG_PM_DEVICE
+#include <zephyr/pm/device.h>
+
+static int lis2dw12_pm_action(const struct device *dev,
+			     enum pm_device_action action)
+{
+	int ret = 0;
+	struct lis2dw12_data *data = dev->data;
+	const struct lis2dw12_device_config *cfg = dev->config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		ret = lis2dw12_set_power_mode(dev, cfg->pm);
+		if (ret) {
+			return ret;
+		}
+		ret = lis2dw12_set_odr(dev, data->odr);
```

Comment:
Why not just 
return lis2dw12_set_odr(dev, data->odr);

As you do in case of error and the default-case?

No strong opinion here, it is just the fact that we are handling it differently that makes me wonder.